Solver/elastic_law.py

```diff
@@ -0,0 +1,310 @@
+import shenfun as sf
+import sympy as sp
+import numpy as np
+import itertools as it
+from shenfun import inner, div, grad, Dx
+
+
+class LinearCauchyElasticity:
+    def __init__(self):
+        self.name = "LinearCauchyElasticity"
+        self.nb_material_parameters = 2
```

Comment:
You could also make these variables private by adding an underscore, i.e., `self._name`.